src/overlays/actors/ovl_En_Ba/z_en_ba.c

```diff
@@ -66,35 +76,450 @@ static ColliderJntSphInit D_809B80D4 = {
+            Matrix_MultVec3f(&sp6C, &this->unk158[i + 1]);
+        }
+        this->unk31A--;
+    } else {
+        Flags_SetSwitch(globalCtx, this->upperParams);
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ba/func_809B71F0.s")
+void EnBa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBa* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ba/func_809B75A0.s")
+    if ((this->actor.params < EN_BA_DEAD_BLOB) && (this->collider.base.acFlags & 2)) {
+        this->collider.base.acFlags &= ~2;
+        this->actor.colChkInfo.health--;
+        if (this->actor.colChkInfo.health == 0) {
+            func_809B75A0(this, globalCtx);
+        } else {
+            func_809B7174(this);
+        }
+    }
+    this->actionFunc(this, globalCtx);
+    if (this->actor.params < EN_BA_DEAD_BLOB) {
+        this->actor.focus.pos = this->unk158[6];
+    }
+    if (this->unk14C >= 2) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ba/func_809B781C.s")
+static Gfx* D_809B8118[] = { 0x060024F0, 0x060027F0, 0x060029F0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ba/EnBa_Update.s")
+void EnBa_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnBa* this = THIS;
+    s32 pad;
+    s16 i;
+    Mtx* mtx = Graph_Alloc(globalCtx->state.gfxCtx, sizeof(Mtx) * 14);
+    Vec3f unused = { 0.0f, 0.0f, 448.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ba/EnBa_Draw.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_ba.c", 933);
+    func_80093D18(globalCtx->state.gfxCtx);
+    if (this->actor.params < EN_BA_DEAD_BLOB) {
+        Matrix_Push();
+        gSPSegment(POLY_OPA_DISP++, 0x0C, mtx);
+        gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(D_809B8118[this->actor.params]));
+        gSPSegment(POLY_OPA_DISP++, 0x09,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 16, 16, 1, 0,
+                                    (globalCtx->gameplayFrames * -10) % 128, 32, 32));
+        for (i = 0; i < 14; i++, mtx++) {
+            Matrix_Translate(this->unk158[i].x, this->unk158[i].y, this->unk158[i].z, MTXMODE_NEW);
+            Matrix_RotateRPY(this->unk2A8[i].x, this->unk2A8[i].y, this->unk2A8[i].z, MTXMODE_APPLY);
+            Matrix_Scale(this->unk200[i].x, this->unk200[i].y, this->unk200[i].z, MTXMODE_APPLY);
+            if ((i == 6) || (i == 13)) {
+                switch (i) {
+                    case 13:
+                        Collider_UpdateSpheres(i, &this->collider);
+                        break;
+                    default:
+                        Matrix_Scale(0.5f, 0.5f, 1.0f, MTXMODE_APPLY);
+                        Collider_UpdateSpheres(8, &this->collider);
+                        break;
+                }
+            }
```

Comment:
Who would do this?

Reply:
I know right? It would clearly be better to have the wrapping `if ((i == 6) || (i == 13))` as a switch with cases 6 and 13 leading to the other switch.

Reply:
It was a pain to match because I didn't want to believe this was how they wrote it... sadly the asm doesn't lie